tests/ui/self/elision/nested-item.rs

```diff
@@ -0,0 +1,26 @@
+// Regression test for #110899.
+// When looking for the elided lifetime for `wrap_ref_Self_Self`,
+// we must not consider the lifetimes on `ref_self` as candidates.
+
+struct Struct;
+
+impl Struct {
+    fn wrap_ref_Self_Self(
+        self: Wrap<
+        //~^ ERROR cannot find type `Wrap` in this scope
+    {
+        fn ref_self(&self, f: &u32) -> &u32 {
+            //~^ ERROR `self` parameter is only allowed in associated functions
+            //~| ERROR can't use generic parameters from outer function
+            f
+        }
+    },
+    Self,
+>,
```

Comment:
With this formatting I don't even understand what happens.
Maybe it's better to use the minimized reproduction from https://github.com/rust-lang/rust/issues/110899#issuecomment-1525745098?